src/main/java/site/coach_coach/coach_coach_server/category/service/CategoryService.java

```diff
@@ -0,0 +1,52 @@
+package site.coach_coach.coach_coach_server.category.service;
+
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import site.coach_coach.coach_coach_server.category.domain.Category;
+import site.coach_coach.coach_coach_server.category.dto.CreateCategoryRequest;
+import site.coach_coach.coach_coach_server.category.repository.CategoryRepository;
+import site.coach_coach.coach_coach_server.common.constants.ErrorMessage;
+import site.coach_coach.coach_coach_server.common.exception.AccessDeniedException;
+import site.coach_coach.coach_coach_server.routine.domain.Routine;
+import site.coach_coach.coach_coach_server.routine.exception.NoExistRoutineException;
+import site.coach_coach.coach_coach_server.routine.repository.RoutineRepository;
+import site.coach_coach.coach_coach_server.routine.service.RoutineService;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryService {
+	private final CategoryRepository categoryRepository;
+	private final RoutineService routineService;
+	private final RoutineRepository routineRepository;
+
+	public Long createCategory(CreateCategoryRequest createCategoryRequest, Long routineId, Long userIdByJwt) {
+		validateAccessCategory(routineId, userIdByJwt);
+
+		Routine routine = Routine.builder()
+			.routineId(routineId)
+			.build();
+
+		Category category = Category.builder()
+			.routine(routine)
+			.categoryName(createCategoryRequest.categoryName())
+			.isCompleted(false)
+			.build();
+
+		return categoryRepository.save(category).getCategoryId();
+	}
+
+	public void validateAccessCategory(Long routineId, Long userIdByJwt) {
+		Routine routine = routineRepository.findById(routineId)
+			.orElseThrow(() -> new NoExistRoutineException(ErrorMessage.NOT_FOUND_ROUTINE));
+
+		if (!routine.getUserId().equals(userIdByJwt)) {
+			Long coachId = routineService.getCoachId(userIdByJwt);
+			if (routine.getCoachId() != coachId) {
+				throw new AccessDeniedException();
+			}
+		} else if (routine.getCoachId() != null) {
+			throw new AccessDeniedException();
+		}
```

Comment:
이건 이전 pr에도.. 남겼었는데 검증하는게.. 가독성이 떨어져서 나누는 게 좋을 것 같아요 아 루틴이랑은 다른 방식이면 그냥 두는 것도 괜찮고요..
음 아니면 
```suggestion
		    if (routine.getUserId().equals(userIdByJwt)) {
        // 일반 사용자로서 자신이 만든 루틴에 접근
        return;
    }

    Long coachId = routineService.getCoachId(userIdByJwt);
    if (routine.getCoachId() != null && routine.getCoachId().equals(coachId)) {
        // 매칭된 코치로서 루틴에 접근
        return;
    }
    throw new AccessDeniedException();
```
이런 방식도 있을 것 같기도 하구요

Reply:
피드백 감사드려요!
일반회원이 코치가 만들어준 루틴에 조회 이외의 다른 용도로는 접근이 안되기때문에 위처럼 코드를 작성했습니다..!
가독성을 최대한 높여보려 했는데 쉽지가 않더라구요 ㅠㅠ